js/app.js

```diff
@@ -44,32 +45,11 @@ async function fetchKontent(api)
 }
 
 
-async function initiateBlogSummaryList()
-{
-    const params = formatString(itemsParams, ["true", pageSize, "post_date[desc]"]);
-    const api = formatString(apiItems, [projectId, params]);
-	const json = await fetchKontent(api);
-
-    if (json.items.length > 0) {
-        var blogSummary = "";
-        for (var i = 0; i < json.pagination.count; i++) {
-            if (json.items[i].system.type === "blog_post") {
-                blogSummary = buildBlogSummary(json.items[i], json.modular_content);
-                document.querySelector("main").appendChild(blogSummary);
-            }
-        }
-        setNextBatchLoad(json.pagination.next_page);
-    } else {
-        var errorMessage = getApiErrorMessage();
-        document.getElementById("loadMoreBlogItems").setAttribute("style", "display:none;");
-        document.querySelector("main").appendChild(errorMessage);
-    }
-}
-
-
-async function loadNextBlogBatch() {
+async function loadBlogItems() {
     var btnLoadMore = document.getElementById("loadMoreBlogItems");
-    const api = btnLoadMore.dataset.next_page;
+    const params = formatString(itemsParams, ["true", pageSize, "post_date[desc]"]) + (tag ? formatString(itemsForTagParams, [tag]) : "");
+    const api = btnLoadMore.dataset.next_page ? btnLoadMore.dataset.next_page : formatString(apiItems, [projectId, params]);
```

Comment:
here you are using a feature of javascript whereby `btnLoadMore.dataset.next_page` evaluates as `truthy` if it exists as a non-empty string. Because of this you can use the coalescing functionality of the `||` operator like this:
```javascript
const api = btnLoadMore.dataset.next_page || formatString(apiItems, [projectId, params]);
```

Reply:
Thank you